### PR TITLE
Introduce changelog process

### PR DESCRIPTION
**Pull request by KonradStaniec**
- modifies `CHANGELOG.md` file to follow https://keepachangelog.com/en/1.0.0/ convention
- add proper instructions in `CONTRIBUTING.md` and `RELEASE_PROCESS.md` files

### CODE REVIEW

File: RELEASE_PROCESS.md [hunk 1]
```diff
@@ -53,12 +56,19 @@ branch (see [CONTRIBUTING.md](./CONTRIBUTING.md#pull-requests)).
...
   * Create new release candidate tags after bugs are fixed.
 * After the team feels the release candidate is ready, create a full release:
   * **Note:** The final release MUST have the same commit hash as the latest
     corresponding release candidate.
   * Create a new annotated git tag in the release branch (follow the [Tagging
     Procedure](#tagging-procedure))
+* After final release is made e.g `v0.Y.0`, backport changelog changes to `main` branch
+  * checkout a new branch from the main branch: `username/backport_changelog`
+  * bring the new section from `release/v0.Y.x` branch to the `CHANGELOG.md` file on `main` branch
+  * open pr against the `main` branch
```
Comment (RafilxTenfen) [inline RELEASE_PROCESS.md:70]:
```suggestion
  * open PR against the `main` branch
```

**Review by maurolacy — APPROVED** [review]
OK with me. This would require more manual work than with automated changelog generation, but the format and the content of the changelog file would be generally better (as long as the manual process is followed, of course).

As part of this, let's then remove the [upate_changelog.sh](https://github.com/babylonlabs-io/babylon/blob/main/scripts/update_changelog.sh) script.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -120,6 +121,25 @@ affect production code).
 If you open a PR in Babylon, it is mandatory to update the relevant
 documentation in `/docs`.
 
+### Updating Changelog
+
+Any pr which is merged to `main` and affects consumers of the codebase,
```
Comment (Lazar955) [inline CONTRIBUTING.md:126]:
```suggestion
Any PR which is merged to `main` and affects consumers of the codebase,
```

File: RELEASE_PROCESS.md [hunk 3]
```diff
@@ -41,6 +41,9 @@ branch (see [CONTRIBUTING.md](./CONTRIBUTING.md#pull-requests)).
   the minor version number, with patch part substituted to `x` (eg: v0.11.x).
   * **PRs targeting directly a release branch can be merged _only_ when
     exceptional circumstances arise**.
+* In the release branch
+  * Create by a PR, new version section in the `CHANGELOG.md` matching the released version e.g
```
Comment (Lazar955) [inline RELEASE_PROCESS.md:45]:
```suggestion
  * Created by a PR, a new version section in the `CHANGELOG.md` matching the released version e.g
```

File: RELEASE_PROCESS.md [hunk 4]
```diff
@@ -53,12 +56,19 @@ branch (see [CONTRIBUTING.md](./CONTRIBUTING.md#pull-requests)).
   * The release verification on devnet must pass.
   * When bugs are found, create a PR for `main`, and backport fixes to the
     release branch.
+  * Before tagging release, create and merge pr to release branch that:
```
Comment (Lazar955) [inline RELEASE_PROCESS.md:59]:
```suggestion
  * Before tagging the release, create and merge PR to the release branch that:
```

**Review by Lazar955 — APPROVED** [review]
LGTM

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -120,6 +121,25 @@ affect production code).
...
+must modify changelog [file](./CHANGELOG.md) accordingly, by adding new entry
+in `Unreleased` section of the changelog.
+
+Examples of changes which require change log update:
+- bug fix
+- adding new api
+- adding new feature
+
+Examples of changed that do not require changelog updates:
```
Comment (gitferry) [inline CONTRIBUTING.md:135]:
```suggestion
Examples of changes that do not require changelog updates:
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -120,6 +121,25 @@ affect production code).
...
+- refactoring of internal implementation
+- adding new tests
+- modifying documentation
+
+The rule of thumb here it that changelog should be updated whenever change is
+visible by external user.
```
Comment (gitferry) [inline CONTRIBUTING.md:141]:
```suggestion
visible by external users.
```

File: RELEASE_PROCESS.md [hunk 7]
```diff
@@ -53,12 +56,19 @@ branch (see [CONTRIBUTING.md](./CONTRIBUTING.md#pull-requests)).
...
+    * Moves all changelog entries form `Unreleased` section of the changelog to newly created section `v0.Y.0`
```
Comment (gitferry) [inline RELEASE_PROCESS.md:60]:
```suggestion
    * Moves all changelog entries form `Unreleased` section of the changelog to the newly created section `v0.Y.0`
```

File: RELEASE_PROCESS.md [hunk 8]
```diff
@@ -53,12 +56,19 @@ branch (see [CONTRIBUTING.md](./CONTRIBUTING.md#pull-requests)).
...
   * Create new release candidate tags after bugs are fixed.
 * After the team feels the release candidate is ready, create a full release:
   * **Note:** The final release MUST have the same commit hash as the latest
     corresponding release candidate.
   * Create a new annotated git tag in the release branch (follow the [Tagging
     Procedure](#tagging-procedure))
+* After final release is made e.g `v0.Y.0`, backport changelog changes to `main` branch
```
Comment (gitferry) [inline RELEASE_PROCESS.md:67]:
```suggestion
* After the final release is made e.g `v0.Y.0`, backport changelog changes to the `main` branch
```

File: RELEASE_PROCESS.md [hunk 9]
```diff
@@ -53,12 +56,19 @@ branch (see [CONTRIBUTING.md](./CONTRIBUTING.md#pull-requests)).
...
+  * checkout a new branch from the main branch: `username/backport_changelog`
+  * bring the new section from `release/v0.Y.x` branch to the `CHANGELOG.md` file on `main` branch
+  * open pr against the `main` branch
+
```
Comment (gitferry) [inline RELEASE_PROCESS.md:71]: blank line?

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -120,6 +121,25 @@ affect production code).
...
+### Updating Changelog
+
+Any pr which is merged to `main` and affects consumers of the codebase,
+must modify changelog [file](./CHANGELOG.md) accordingly, by adding new entry
+in `Unreleased` section of the changelog.
+
+Examples of changes which require change log update:
+- bug fix
+- adding new api
+- adding new feature
```
Comment (gitferry) [inline CONTRIBUTING.md:133]:
```suggestion
- bug fixes
- adding new APIs
- adding new features
```

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -120,6 +121,25 @@ affect production code).
...
+
+Examples of changed that do not require changelog updates:
+- refactoring of internal implementation
+- adding new tests
+- modifying documentation
+
+The rule of thumb here it that changelog should be updated whenever change is
```
Comment (gitferry) [inline CONTRIBUTING.md:140]:
```suggestion
The rule of thumb here is that the changelog should be updated as long as the is
```

**Review by gitferry — APPROVED** [review]
LGTM! Some nits:

**Pull request merged**